index.js

```diff
@@ -117,6 +121,39 @@ f1.prototype = {
    */
   targets: function( targets ) {
 
+    var instance = this;
+
+    if (targets) {
+      _.forEach(targets, function (el, elementId) {
+        var elementData = {
+          id: elementId,
+          get states() {
+            return _.keys(instance.defStates).reduce(function (dict, key) {
+              dict[key] = instance.defStates[key][elementId];
+              return dict;
+            }, {});
+          },
+          set states(newStates) {
+            // TODO: what to do about state deletions? this only handles modifications or new states
+            Object.keys(newStates).forEach(function (stateId) {
+              instance.defStates[stateId][elementId] = newStates[stateId];
+            });
+            instance.go(instance.state);
+          },
+          get transitions() {
+            // TODO. Note: by default it's in the form of [{animation: {elementId1: { properties }, anotherElement: { properties }}, from, to}]
+            // should change it to [{animation: { properties }, from, to}]
+            return [];
+          },
+          $go: instance.go.bind(instance)
+        };
+        // add to global so the devtool can access the list of elements
+        global.f1targets[elementId] = elementData;
```

Comment:
Keep in mind though that the extension's javascript executes in a completely different context and only has access to the main page's root context (i.e. `window`), and could potentially run after the page's script has finished executing (so can't modify f1 behaviour inside the extension)

f1 would need to provide some way for its contents to be accessible via the global scope if we want to be able to generate a list of all the f1 elements on the page.

Another thing we could do is to walk the DOM tree and find all elements with a tag (like an `is-f1` attribute or something) but that's essentially adding globals in a more roundabout way